src/methods/sep8Send/sendActionRequiredFields.ts

```diff
@@ -0,0 +1,68 @@
+import { log } from "helpers/log";
+import {
+  Sep8ActionRequiredSendParams,
+  Sep8ActionRequiredSentResult,
+} from "types/types.d";
+
+export const sendActionRequiredFields = async ({
+  actionFields,
+  actionMethod,
+  actionUrl,
+}: Sep8ActionRequiredSendParams): Promise<Sep8ActionRequiredSentResult> => {
+  log.request({
+    title: `Sending action required fields to SEP-8 server with \`${actionMethod} ${actionUrl}\``,
+    body: actionFields,
+  });
+  const sep8ActionRequiredResult = await fetch(actionUrl, {
+    method: actionMethod,
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(actionFields),
+  });
+  const resultJson = await sep8ActionRequiredResult.json();
+
+  let validatedResponse: Sep8ActionRequiredSentResult;
+  switch (resultJson.result) {
+    case "no_further_action_required":
+      validatedResponse = {
+        result: "no_further_action_required",
+      };
+      break;
+
+    case "follow_next_url":
+      if (!resultJson.next_url) {
+        throw new Error(`Missing "next_url" parameter`);
```

Comment:
❓ What is the grammatical error message conventions for js?
Should there be a period after the message if the starting letter is capitalized?
Should it be all lowercase if its an internal error?

Reply:
In this application an error thrown that way will be directly presented to the user so it should be in a readable format.

Regarding conventions like "not starting with a capital letter" or "not ending with punctuation", they make sense in Go because of the way the Go errors are prepended/wrapped... a sequence of errors in go will be printed like `"errStr1: errStr2: errStr3"` but that does not apply for javascript nor typescript.

Here are a couple of references in the subject:
* https://github.com/golang/go/wiki/CodeReviewComments#error-strings
* https://jeremybytes.blogspot.com/2021/01/go-golang-error-handling-different.html